src/System.Management.Automation/engine/regex.cs

```diff
@@ -233,33 +222,34 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
             Span<char> temp = pattern.Length < StackAllocThreshold ? stackalloc char[pattern.Length * 2 + 1] : new char[pattern.Length * 2 + 1];
             int tempIndex = 0;
 
-            for (int i = 0; i < pattern.Length; i++)
+            if (escapeOnlyBrackets)
             {
-                char ch = pattern[i];
-
-                //
-                // if it is a wildcard char, escape it
-                //
-                if (IsWildcardChar(ch) && !charsNotToEscape.Contains(ch))
+                for (int i = 0; i < pattern.Length; i++)
                 {
-                    temp[tempIndex++] = escapeChar;
-                }
-
-                temp[tempIndex++] = ch;
-            }
-
-            string s = null;
+                    char ch = pattern[i];
+                    if (ch == '[' || ch == ']')
```

Comment:
```suggestion
                    if (ch is '[' or ']')
```

Reply:
@xtqqczze The PR is only for removing Utils.Separators.StarOrQuestion. No plans for style changes.